api/edge_api/identities/traits/permissions.py

```diff
@@ -0,0 +1,17 @@
+from rest_framework.permissions import IsAuthenticated
+
+from environments.models import Environment
+from environments.permissions.constants import VIEW_ENVIRONMENT
+
+
+class EnvironmentIdentityTraitsPermission(IsAuthenticated):
+    def has_permission(self, request, view):
+        if not super().has_permission(request, view):
+            return False
+
+        environment = Environment.objects.get(
+            api_key=view.kwargs.get("environment_api_key")
+        )
+        if not request.user.has_environment_permission(VIEW_ENVIRONMENT, environment):
+            return False
+        return request.user.is_environment_admin(environment)
```

Comment:
The above call to `has_environment_permission` will check if they're an environment admin so we should just simplify these three lines to: 

```python
return request.user.has_environment_permission(VIEW_ENVIRONMENT, environment)
```